cmake/kokkos_tpls.cmake

```diff
@@ -60,6 +60,8 @@ SET(HPX_DEFAULT OFF)
 ENDIF()
 KOKKOS_TPL_OPTION(HPX ${HPX_DEFAULT})
 
+KOKKOS_TPL_OPTION(THREADS ${Kokkos_ENABLE_THREADS} TRIBITS Pthread)
```

Comment:
I thought @ndellingwood was saying that the `Thread` backed should bot be enabled/disabled based on `Trililinos`' `Pthread` TPL.

Reply:
@masterleinad : Ready to merge?

Reply:
I don't have the power to do that but I'm happy to approve (assuming @ndellingwood verified that this works correctly with `Trilinos`).